### PR TITLE
Implement limitations on maximum collection size

### PR DESCRIPTION
**Pull request by fyrchik**
Arrays, Structs and Maps have maximum size defined as MaxArraySize.
We need to return an error in case collection becomes too big.

### CODE REVIEW

File: pkg/vm/vm.go [hunk 1]
```diff
@@ -786,10 +786,16 @@ func (v *VM) execute(ctx *Context, op Instruction, parameter []byte) {
 		switch t := arrElem.value.(type) {
 		case *ArrayItem:
```
Comment (volekerb) [inline pkg/vm/vm.go]: can't we combine these 2 parts of code: 787-790 and 791-795? they look identical

Comment (roman-khimov) [reply pkg/vm/vm.go]: Nope, it's Go and even though you can do something like
```go
switch t := arrElem.value.(type) {
		case *ArrayItem, *StructItem:
```
The rest of it just won't work this way. Like compare this: https://play.golang.org/p/FbHgUiPRAsX and this: https://play.golang.org/p/dpAQC5zzDAs

**Comment by fyrchik** [comment]
> I guess it's a part of #373, but not yet a full fix for it, right?

Yes, exactly. Forgot to mention the issue.

File: pkg/vm/vm.go (hunk 1 above)
Comment (fyrchik) [reply pkg/vm/vm.go]: They _are_ identical, however we need to be able to access a field in a type-switched structure (`t.value = ...`).

**Pull request merged**